File: pinyougou_manager_web/src/main/webapp/js/service/specificationService.js
app.service('specificationService',function ($http) {
    // 条件查询
   this.search=function (entity,pageNum,pageSize) {
       return $http.post('../specification/search.do?pageNum='+pageNum+'&pageSize='+pageSize,entity);
   }
    //增加规格
   this.add= function (entity) {
       return $http.post('../specification/add.do',entity);
   }
   // 根据ID查询
    this.findOneById= function (id) {
        return $http.get('../specification/findOneById.do?id='+id);
    }
    //修改规格
    this.update=function (entity) {
        return  $http.post('../specification/update.do',entity);
    }
    //删除规格
    this.dele=function (ids) {
        return  $http.get('../specification/delete.do?ids='+ids);
    }
    //查询规格列表
    this.findSpecList=function () {
        return $http.get('../specification/findSpecList.do');
    }
});

File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.pojo.TbSpecificationOption;
import entity.PageResult;
import groupEntity.Specification;

import java.util.List;
import java.util.Map;

/**
 * @author qinshiji
 * @data 2019/1/11 15:26
 */
public interface SpecificationService {
    /**
     * 条件查询
     * @param specification
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult search(TbSpecification specification, Integer pageNum, Integer pageSize);

    /**
     * 新增规格
     * @param specification
     */
    void add(Specification specification);

    /**
     * 根据ID查询
     * @param id
     * @return
     */
    Specification findOneById(Long id);

    /**
     * 更改规格
     * @param specification
     */
    void update(Specification specification);

    /**
     * 删除规格
     * @param ids
     */
    void delete(Long[] ids);

    List<Map> findSpecList();
}
